xcdat/bounds.py

```diff
@@ -374,15 +477,18 @@ def _create_bounds(self, axis: CFAxisKey, coord_var: xr.DataArray) -> xr.DataArr
         diffs = np.insert(diffs, 0, diffs[0])
         diffs = np.append(diffs, diffs[-1])
 
-        # `cftime` objects only support arithmetic using `timedelta` objects, so
-        # the values of  `diffs` must be casted from `dtype="timedelta64[ns]"`
-        # to `timedelta` objects.
-        if axis == "T" and issubclass(type(coord_var.values[0]), cftime.datetime):
-            bounds = create_time_bounds(coord_var)
-            return bounds
-
-        # width parameter: determines bounds location relative to midpoints
-        width = 0.5
+        if axis == "T":
+            if not contains_datetime_like_objects(coord_var):
+                raise TypeError(
+                    f"Bounds cannot be created for '{coord_var.name}' coordinates "
+                    "because it is not decoded as `cftime.datetime` or `np.datetime`. "
+                    f"Try decoding '{coord_var.name}' first then adding bounds."
+                )
+            # `cftime` objects only support arithmetic using `timedelta` objects, so
+            # the values of  `diffs` must be casted from `dtype="timedelta64[ns]"`
+            # to `timedelta` objects.
+            if contains_cftime_datetimes(xr.as_variable(coord_var)):
+                diffs = pd.to_timedelta(diffs)
```

Comment:
I wonder if we should require this for time axes in the generic `create_bounds` function. If the user opened the dataset without decoding then the time values would be floats/ints (e.g., hours since... with values like [0, 30, 60, ...]) and this function could come up with reasonable (un-decoded) bounds. I think this requirement could be eliminated.

Reply:
Thanks for catching this. I accidentally created a regression from the `main` code because we supported non-decoded time coordinates before. I'll delete this line.

Reply:
Good catch, we previously supported numerically encoded time coordinates so this conditional creates a new regression. I'll remove this conditional.